core/src/graph/graph.rs

```diff
@@ -28,7 +27,7 @@ pub struct Graph {
 	user_id: DsnpUserId,
 	schema_id: SchemaId,
 	pages: PageMap,
-	user_key_manager: Rc<RefCell<UserKeyManager>>,
+	user_key_manager: Rc<RefCell<dyn UserKeyManagerBase>>,
```

Comment:
Allows injecting Mock. Didn't use generic `Graph<E>` since it the code change would be much more than desired.

Reply:
💯